src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp

```diff
@@ -283,64 +285,31 @@ void S3ObjectStorage::removeObjects(const std::vector<std::string> & paths)
             request.SetBucket(bucket);
             request.SetDelete(delkeys);
             auto outcome = client_ptr->DeleteObjects(request);
-            throwIfError(outcome);
+
+            if (!outcome.IsSuccess() && (!if_exists || !isNotFoundError(outcome.GetError().GetErrorType())))
+                throwIfError(outcome);
```

Comment:
Also, in this case even if absence of key may be ok for us, the log will be polluted with error messages from aws sdk. I don't know how to supress them.

Reply:
It may also make sense to have l. 289-290 resp. 217-218 as a separate function.